pkg/network/stream/stream.go

```diff
@@ -0,0 +1,34 @@
+package stream
+
+import (
+	"time"
+
+	"github.com/UnnecessaryRain/ironway-core/pkg/network/protocol"
+)
+
+// Stream is a channel like stream to
+type Stream struct {
```

Comment:
Stream should be an interface, and a Chat struct should implement it's functions